### ISSUE TITLE
Verify that App Services slot swaps are being detected in Web SDK

### ISSUE THREAD

**Issue by d3r3kk** [Microsoft/ApplicationInsights-dotnet-server#872]
There is some concern that slot swap detection is not reliable in the Web SDK for App Services. We have seen evidence that the slot swap was not being detected properly in Azure Functions, and we must ensure that other App Services (Web Apps, for instance) are not affected in the same way.

See comments in PR #870 regarding the use of WEBSITE_HOSTNAME for further insight & details.

## Repro Steps
1. Create test web application in Azure (use the New Devops Project workflow)
2. Add a deployment slot to the application's CD pipeline
3. Instrument the application with the latest beta Web SDK
4. Perform a slot swap and watch App Insights Analytics to ensure WEBSITE_HOSTNAME is being recorded properly.

##  Actual Behavior
**TO BE VERIFIED**

## Expected Behavior
The nodeName property of application insights telemetry will update to reflect the correct slot name within 30 seconds of the slot change.

## Version Info 
SDK Version : latest beta.
.NET Version :	4.5 (.NET full framework) or 1.3 (.NET Standard) 						
Hosting Info (IIS/Azure WebApps/ etc) : WebApps

**Comment by d3r3kk** [comment]
See PR Azure/azure-webjobs-sdk#1382 for further detail.

**Comment by MS-TimothyMothra** [comment]
@d3r3kk will you be able to test this next week?
Please follow up with @Dmitry-Matveev

**Comment by d3r3kk** [comment]
Will investigate by EOD tomorrow.

**Comment by d3r3kk** [comment]
# Environment values investigation

*Can we use the environment to detect slot-swaps in Application Insights?*

**tl;dr** No, we cannot. The environment variables on the host are not propagated to the process after a slot swap.

## Findings at Runtime

### What does the environment say in a demo app?

The environment variables are not updating how we need them to, in order to detect slot - swaps.
I wrote a demo ASP.NET app to show the current values of the environment variables before/after a slot swap.

Here's the variables in the 'dev slot' when I deploy my app into the DEV environment in my release definition in VSTS:

**Environment Variable Name** | **Value**
------------------------------- | ----------------------------
WEBSITE_SITE_NAME | dek-app2infra
WEBSITE_HOME_STAMPNAME | waws-prod-sn1-115
WEBSITE_HOSTNAME | **dek-app2infra-dev.azurewebsites.net**
WEBSITE_OWNER_NAME | aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee+dekapp2infra9700-SouthCentralUSwebspace
WEBSITE_SLOT_NAME | 

And here's the values again after swapping the slots, my demo app is now in the 'production' slot:

**Environment Variable Name** | **Value**
------------------------------- | ----------------------------
WEBSITE_SITE_NAME | dek-app2infra
WEBSITE_HOME_STAMPNAME | waws-prod-sn1-115
WEBSITE_HOSTNAME | **dek-app2infra-dev.azurewebsites.net**
WEBSITE_OWNER_NAME | aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee+dekapp2infra9700-SouthCentralUSwebspace
WEBSITE_SLOT_NAME | 

Notice that the WEBSITE_HOSTNAME is the url to the *dev slot* of the site I created, and that no value changed at all. 

## Peek Under the Hood

What is very odd to me, is that the underlying system seems to have *different values* for these variables! Here's the values from both of the slot's kudu site, from the Environment Variables section:

Environment Variable | Dev Slot Value | Production Slot Value
------------------------ | ---------------- | ------------------------
WEBSITE_SITE_NAME | dek-app2infra | dek-app2infra
WEBSITE_HOME_STAMPNAME | waws-prod-sn1-115 | waws-prod-sn1-115
WEBSITE_HOSTNAME | dek-app2infra-dev.azurewebsites.net | dek-app2infra.azurewebsites.net
WEBSITE_OWNER_NAME | aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee+dekapp2infra9700-SouthCentralUSwebspace | aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee+dekapp2infra9700-SouthCentralUSwebspace
WEBSITE_SLOT_NAME |  | 

## Summary

So our current attempt to fix up this issue, inspired in part by the Azure Functions team's use of a special environment variable 'WEBSITE_SLOT_NAME' didn't pan out. No environment variable is currently giving us any indication of the slot the app is running in, and we will have to use some other method to detect when this happens.

## TODO

Ask the Antares team if they plan on supporting the update of the environment variables we want to rely on for this.

**If environment variable updates won't get updated:** Remove the Environment variable monitor & subscription model, and investigate further to find a better way to detect slot swaps. Perhaps using request headers or some other more definitive way exists (we could repurpose the environment variable monitor to be a request-header-value monitor).

**Comment by d3r3kk** [comment]
Note that this does not impede the app2infra scenario for Web Apps.
